icu4c/source/common/patternprops.cpp

```diff
@@ -173,6 +173,15 @@ PatternProps::skipWhiteSpace(const UChar *s, int32_t length) {
     return s;
 }
 
+int32_t
+PatternProps::skipWhiteSpace(const UnicodeString& s, int32_t start) {
+    int32_t i = start;
+    while(i<s.length() && isWhiteSpace(s.charAt(i))) {
```

Comment:
Please pull s.length() out into
    int32_t length = s.length();

--> UnicodeString.length() is cheap but not free.